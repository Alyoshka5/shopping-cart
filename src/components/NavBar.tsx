import { NavLink, useLocation } from 'react-router-dom';
import '../styles/NavBar.css';
import Icon from '@mdi/react';
import { mdiChip } from '@mdi/js';

export default function NavBar() {
    const currentPath = useLocation().pathname;
    const currentPathClassName = currentPath === '/' ? 'home-path' : 'shop-path';

    return (
        <nav className={`${currentPathClassName}-nav`}>
            <ul>
                <div className='company-icon'>
                    <h2>TechHub</h2>
                    <Icon path={mdiChip} size={1.5} />
                </div>

                <li><NavLink to='/' className={`${currentPathClassName}-link`}>Home</NavLink></li>
                <li><NavLink to='/shop' className={`${currentPathClassName}-link`}>Shop</NavLink></li>
            </ul>
        </nav>
    )
}