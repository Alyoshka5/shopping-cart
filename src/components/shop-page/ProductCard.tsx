import { ProductCardProps } from "../../types"
import '../../styles/ProductCard.css';
import Icon from '@mdi/react';
import { mdiStar, mdiStarOutline } from '@mdi/js';
import { useState } from "react";
import QuantityCounter from "./QuantityCounter";

export default function ProductCard({ product, productId, setShoppingCart }: ProductCardProps) {

    const [quantity, setQuantity] = useState(1);

    const starArray = Array.from({ length: product.rating });
    const emptyStarArray = Array.from({ length: 5 - product.rating });

    const productName = product.name.length > 60 ? `${product.name.substring(0, 60)}...` : product.name;

    return (
        <div className="product-card">
            <div className="img-container">
                <img src={product.imageUrl} alt="" />
            </div>
            <div className="product-info-cart-button-container">
                <div className="product-info">
                    <div className="name-price-container">
                        <div className="name">{productName}</div>
                        <div className="price">${product.price}</div>
                    </div>
                    <div className="company">{product.company}</div>
                    <div className="rating">
                        {starArray.map((_, idx) => <Icon path={mdiStar} size={1} key={idx} />)}
                        {emptyStarArray.map((_, idx) => <Icon path={mdiStarOutline} size={1} key={idx} />)}
                    </div>
                </div>
                <div className="add-to-cart-container">
                    <button className="add-to-cart-button">Add to Cart</button>
                    <QuantityCounter quantity={quantity} setQuantity={setQuantity} />
                </div>
            </div>
        </div>
    )
}