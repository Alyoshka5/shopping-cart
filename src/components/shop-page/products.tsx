const products = [
    {
      name: "Logitech MX Mechanical Keyboard",
      price: 29.99,
      rating: 4,
      company: "Logitech",
      description: [
        "Mechanical keyboard with customizable RGB lighting",
        "Durable and responsive keys for comfortable typing",
        "Wired connection for reliable performance",
        "Compatible with Windows and macOS",
      ],
      imageUrl: "https://m.media-amazon.com/images/I/61ge3d+MQwL._AC_UF894,1000_QL80_.jpg"
    },
    {
      name: "MacBook Air 13\"",
      price: 999.99,
      rating: 5,
      company: "Apple",
      description: [
        "Thin and lightweight design with high-resolution Retina display",
        "Powerful processor and ample storage for seamless performance",
        "Long battery life for all-day usage",
        "Advanced security features and intuitive operating system",
      ],
      imageUrl: "https://m.media-amazon.com/images/I/51dafnlz6wL.jpg"
    },
    {
      name: "Lumiman LED Lightbulb",
      price: 12.99,
      rating: 3,
      company: "Lumiman",
      description: [
        "Smart LED lightbulb with customizable RGB colors",
        "Compatible with voice assistants for easy control",
        "Adjustable brightness and scheduling options",
        "Energy-efficient and long-lasting",
      ],
      imageUrl: "https://m.media-amazon.com/images/I/510rNVKkZNL.jpg"
    },
    {
      name: "Amazon Alexa - Echo Dot",
      price: 49.99,
      rating: 4,
      company: "Amazon",
      description: [
        "Smart voice assistant with built-in speakers",
        "Plays music, answers questions, and controls smart home devices",
        "Voice recognition for personalized responses",
        "Supports various entertainment and productivity apps",
      ],
      imageUrl: "https://pisces.bbystatic.com/image2/BestBuy_US/images/products/6287/6287974_sd.jpg"
    },
    {
      name: "XPRIT 6.5\" Hoverboard",
      price: 199.99,
      rating: 4,
      company: "Segway",
      description: [
        "Self-balancing electric scooter with hoverboard design",
        "Intuitive controls and smooth maneuverability",
        "Long battery life and fast charging",
        "Built-in Bluetooth speakers for music streaming",
      ],
      imageUrl: "https://m.media-amazon.com/images/I/61tC4ZrTiDL._AC_UF894,1000_QL80_.jpg"
    },
    {
      name: "Dell S2421HS 24\" Full HD Monitor",
      price: 249.99,
      rating: 4,
      company: "Dell",
      description: [
        "High-resolution display with vibrant colors",
        "Wide viewing angles for immersive visual experience",
        "Multiple connectivity options for versatile usage",
        "Ergonomic design for comfortable viewing",
      ],
      imageUrl: "https://m.media-amazon.com/images/I/81vf4znYbaL.jpg"
    },
];

export default products;