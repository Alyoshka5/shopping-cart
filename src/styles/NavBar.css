nav {
    width: 100%;
    height: 4.6rem;
    padding: 0 7rem 0 5rem;
    box-sizing: border-box;
    overflow: hidden;
    position: fixed;
    top: 0;
    z-index: 5;
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.shop-path-nav {
    background-color: #fff;
    box-shadow: 0 0 10px 0 #00000080;
}

ul {
    list-style-type: none;
    margin: 0;
    padding: 1rem 1rem;
    overflow: hidden;
    display: flex;
    align-items: center;
    gap: 4rem;
}

.company-icon {
    font-size: 1.4rem;
    color: var(--theme-color);
    display: flex;
    justify-content: center;
    align-items: center;
}
.company-icon h2 {
    margin: 0;
}

a {
    font-size: 1.2rem;
    text-decoration: none;
    text-transform: uppercase;
}
li a:hover {
    color: var(--theme-color);
}

.home-path-link {
    color: #fff;
}

.shop-path-link {
    color: #000;
}