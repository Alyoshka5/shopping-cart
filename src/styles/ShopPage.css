.shop-page {
    height: 100%;
    padding: 8rem 5%;
    display: flex;
    justify-content: center;
}

.shop-page .products {
    display: flex;
    gap: 1.5rem;
    flex-wrap: wrap;
}
