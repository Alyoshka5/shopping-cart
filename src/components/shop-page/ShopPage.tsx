import products from './products';
import ProductCard from './ProductCard';
import '../../styles/ShopPage.css';

export default function ShopPage() {

    return (
        <div className="shop-page">
            <div className='products'>
                { products.map((product, productId) => {
                    return (
                        <ProductCard product={product} productId={productId} key={productId} />
                    )
                })}
            </div>
        </div>
    )
}