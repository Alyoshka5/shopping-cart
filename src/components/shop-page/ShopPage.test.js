import rendrer from 'react-test-renderer';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import ShopPage from './ShopPage';
import NavBar from '../NavBar';
import ProductCard from './ProductCard';
import products from './products';
import { render, screen } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import { act } from 'react-dom/test-utils';

it('renders shop page content', () => {
    const pageTree = rendrer.create(<BrowserRouter><ShopPage /></BrowserRouter>).toJSON();

    expect(pageTree).toMatchSnapshot();
});

describe('nav bar displays item count', () => {
    it('renders 0 on item count of cart icon when cart is empty', () => {
        const cart = [];

        render(<MemoryRouter initialEntries={['/shop']}><NavBar shoppingCart={cart} /></MemoryRouter>);
        const itemCount = screen.getByTestId('cart-item-count');

        expect(itemCount.textContent).toBe('0');
    });

    it('renders 3 on item count of cart icon when cart has 3 items', () => {
        const cart = {1: 2, 2: 1};

        render(<MemoryRouter initialEntries={['/shop']}><NavBar shoppingCart={cart} /></MemoryRouter>);
        const itemCount = screen.getByTestId('cart-item-count');

        expect(itemCount.textContent).toBe('3');
    });
});

describe('adding items to cart', () => {
    it('adds 1 product to cart', async () => {
        const user = userEvent.setup();
        let cart = {};
        const mockSetShoppingCart = jest.fn((updateFunction) => {
            cart = updateFunction(cart);
        });

        render(<BrowserRouter><ProductCard product={products[1]} productId={1} setShoppingCart={mockSetShoppingCart} /></BrowserRouter>);
        render(<MemoryRouter initialEntries={['/shop']}><NavBar shoppingCart={cart} /></MemoryRouter>);
        const cartButton = screen.getByRole('button', { name: 'Add to Cart' });

        await act(async () => {
            await user.click(cartButton);
        });

        expect(cart).toMatchObject({1: 1});
    });

    it('adds 2 products to cart', async () => {
        const user = userEvent.setup();
        let cart = {};
        const mockSetShoppingCart = jest.fn((updateFunction) => {
            cart = updateFunction(cart);
        });

        render(<BrowserRouter><ProductCard product={products[0]} productId={0} setShoppingCart={mockSetShoppingCart} /></BrowserRouter>);
        render(<BrowserRouter><ProductCard product={products[1]} productId={1} setShoppingCart={mockSetShoppingCart} /></BrowserRouter>);
        render(<MemoryRouter initialEntries={['/shop']}><NavBar shoppingCart={cart} /></MemoryRouter>);
        const cartButtons = screen.getAllByRole('button', { name: 'Add to Cart' });

        await act(async () => {
            await user.click(cartButtons[0]);
            await user.click(cartButtons[1]);
        });

        expect(cart).toMatchObject({0: 1, 1: 1});
    });

    it('adds multiple items of same product to cart', async () => {
        const user = userEvent.setup();
        let cart = {};
        const mockSetShoppingCart = jest.fn((updateFunction) => {
            cart = updateFunction(cart);
        });

        render(<BrowserRouter><ProductCard product={products[1]} productId={1} setShoppingCart={mockSetShoppingCart} /></BrowserRouter>);
        render(<MemoryRouter initialEntries={['/shop']}><NavBar shoppingCart={cart} /></MemoryRouter>);
        const cartButton = screen.getByRole('button', { name: 'Add to Cart' });
        const incrementButton = screen.getByRole('button', { name: '+' });

        await act(async () => {
            await user.click(incrementButton);
        });
        await act(async () => {
            await user.click(incrementButton);
        });
        await act(async () => {
            await user.click(cartButton);
        });

        expect(cart).toMatchObject({1: 3});
    });

    it('resets quantity counter to 1 after add to cart button click', async () => {
        const user = userEvent.setup();
        let cart = [];
        const mockSetShoppingCart = jest.fn((updateFunction) => {
            cart = updateFunction(cart);
        });

        render(<BrowserRouter><ProductCard product={products[1]} productId={1} setShoppingCart={mockSetShoppingCart} /></BrowserRouter>);
        render(<MemoryRouter initialEntries={['/shop']}><NavBar shoppingCart={cart} /></MemoryRouter>);
        const cartButton = screen.getByRole('button', { name: 'Add to Cart' });
        const incrementButton = screen.getByRole('button', { name: '+' });
        const quantityValue = screen.getByTestId('quantity-value');

        await act(async () => {
            await user.click(incrementButton);
        });
        expect(quantityValue.textContent).toMatch('2');
        
        await act(async () => {
            await user.click(cartButton);
        });
        expect(quantityValue.textContent).toMatch('1');

    });
});