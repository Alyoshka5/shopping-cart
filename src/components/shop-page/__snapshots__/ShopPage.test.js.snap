// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders shop page content 1`] = `
<div
  className="shop-page"
>
  <div
    className="products"
  >
    <div
      className="product-card"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          src="https://m.media-amazon.com/images/I/61ge3d+MQwL._AC_UF894,1000_QL80_.jpg"
        />
      </div>
      <div
        className="product-info-cart-button-container"
      >
        <div
          className="product-info"
        >
          <div
            className="name-price-container"
          >
            <div
              className="name"
            >
              Logitech MX Mechanical Keyboard
            </div>
            <div
              className="price"
            >
              $
              29.99
            </div>
          </div>
          <div
            className="company"
          >
            Logitech
          </div>
          <div
            className="rating"
          >
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
          </div>
        </div>
        <div
          className="add-to-cart-container"
        >
          <button
            className="add-to-cart-button"
            onClick={[Function]}
          >
            Add to Cart
          </button>
          <div
            className="quantity-counter"
          >
            <button
              className="decrement-button"
              onClick={[Function]}
            >
              -
            </button>
            <div
              className="quantity-value"
            >
              1
            </div>
            <button
              className="increment-button"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          src="https://m.media-amazon.com/images/I/51dafnlz6wL.jpg"
        />
      </div>
      <div
        className="product-info-cart-button-container"
      >
        <div
          className="product-info"
        >
          <div
            className="name-price-container"
          >
            <div
              className="name"
            >
              MacBook Air 13"
            </div>
            <div
              className="price"
            >
              $
              999.99
            </div>
          </div>
          <div
            className="company"
          >
            Apple
          </div>
          <div
            className="rating"
          >
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
          </div>
        </div>
        <div
          className="add-to-cart-container"
        >
          <button
            className="add-to-cart-button"
            onClick={[Function]}
          >
            Add to Cart
          </button>
          <div
            className="quantity-counter"
          >
            <button
              className="decrement-button"
              onClick={[Function]}
            >
              -
            </button>
            <div
              className="quantity-value"
            >
              1
            </div>
            <button
              className="increment-button"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          src="https://m.media-amazon.com/images/I/510rNVKkZNL.jpg"
        />
      </div>
      <div
        className="product-info-cart-button-container"
      >
        <div
          className="product-info"
        >
          <div
            className="name-price-container"
          >
            <div
              className="name"
            >
              Lumiman LED Lightbulb
            </div>
            <div
              className="price"
            >
              $
              12.99
            </div>
          </div>
          <div
            className="company"
          >
            Lumiman
          </div>
          <div
            className="rating"
          >
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
          </div>
        </div>
        <div
          className="add-to-cart-container"
        >
          <button
            className="add-to-cart-button"
            onClick={[Function]}
          >
            Add to Cart
          </button>
          <div
            className="quantity-counter"
          >
            <button
              className="decrement-button"
              onClick={[Function]}
            >
              -
            </button>
            <div
              className="quantity-value"
            >
              1
            </div>
            <button
              className="increment-button"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          src="https://pisces.bbystatic.com/image2/BestBuy_US/images/products/6287/6287974_sd.jpg"
        />
      </div>
      <div
        className="product-info-cart-button-container"
      >
        <div
          className="product-info"
        >
          <div
            className="name-price-container"
          >
            <div
              className="name"
            >
              Amazon Alexa - Echo Dot
            </div>
            <div
              className="price"
            >
              $
              49.99
            </div>
          </div>
          <div
            className="company"
          >
            Amazon
          </div>
          <div
            className="rating"
          >
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
          </div>
        </div>
        <div
          className="add-to-cart-container"
        >
          <button
            className="add-to-cart-button"
            onClick={[Function]}
          >
            Add to Cart
          </button>
          <div
            className="quantity-counter"
          >
            <button
              className="decrement-button"
              onClick={[Function]}
            >
              -
            </button>
            <div
              className="quantity-value"
            >
              1
            </div>
            <button
              className="increment-button"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          src="https://m.media-amazon.com/images/I/61tC4ZrTiDL._AC_UF894,1000_QL80_.jpg"
        />
      </div>
      <div
        className="product-info-cart-button-container"
      >
        <div
          className="product-info"
        >
          <div
            className="name-price-container"
          >
            <div
              className="name"
            >
              XPRIT 6.5" Hoverboard
            </div>
            <div
              className="price"
            >
              $
              199.99
            </div>
          </div>
          <div
            className="company"
          >
            Segway
          </div>
          <div
            className="rating"
          >
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
          </div>
        </div>
        <div
          className="add-to-cart-container"
        >
          <button
            className="add-to-cart-button"
            onClick={[Function]}
          >
            Add to Cart
          </button>
          <div
            className="quantity-counter"
          >
            <button
              className="decrement-button"
              onClick={[Function]}
            >
              -
            </button>
            <div
              className="quantity-value"
            >
              1
            </div>
            <button
              className="increment-button"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <div
        className="img-container"
      >
        <img
          alt=""
          src="https://m.media-amazon.com/images/I/81vf4znYbaL.jpg"
        />
      </div>
      <div
        className="product-info-cart-button-container"
      >
        <div
          className="product-info"
        >
          <div
            className="name-price-container"
          >
            <div
              className="name"
            >
              Dell S2421HS 24" Full HD Monitor
            </div>
            <div
              className="price"
            >
              $
              249.99
            </div>
          </div>
          <div
            className="company"
          >
            Dell
          </div>
          <div
            className="rating"
          >
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
            <svg
              role="presentation"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z"
                style={
                  Object {
                    "fill": "currentColor",
                  }
                }
              />
            </svg>
          </div>
        </div>
        <div
          className="add-to-cart-container"
        >
          <button
            className="add-to-cart-button"
            onClick={[Function]}
          >
            Add to Cart
          </button>
          <div
            className="quantity-counter"
          >
            <button
              className="decrement-button"
              onClick={[Function]}
            >
              -
            </button>
            <div
              className="quantity-value"
            >
              1
            </div>
            <button
              className="increment-button"
              onClick={[Function]}
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
